pkg/aggregator/internal/tags/store.go

```diff
@@ -28,9 +30,36 @@ type Entry struct {
 	refs atomic.Uint64
 
 	// tags contains the cached tags in this entry.
-	tags []string
+	tags tagSet
+}
+
+type tagSet []string
+
+// SizeInBytes returns the size of the tagSet in bytes.
+func (t tagSet) SizeInBytes() int {
+	return util.SizeOfStringSlice(t)
+}
+
+// DataSizeInBytes returns the size of the tagSet data in bytes.
+func (t tagSet) DataSizeInBytes() int {
+	// This could be optimized by keeping track of the size when the TagSet is created.
```

Comment:
As an aside, if you want to optimize this, you could track of bytes in the store as entries are added and removed, instead of recomputing in `visit`. This way there is no need to store extra data in the `Entry` or the `tagSet`.

Reply:
I looked at the code again, and this seems cheap enough currently to leave it this way no?
I was trying to keep the telemetry code in a single place, as it is today